### PR TITLE
[Docs] Add more details on INSERTs

### PR DESCRIPTION
**Pull request by justindeguzman**
### Changelog category (leave one):
- Documentation (changelog entry is not required)

### Changelog entry (a user-readable short description of the changes that goes to CHANGELOG.md):
Add more details on INSERTs regarding replicated setups, long-running inserts, and asynchronous inserts

### CODE REVIEW

**Comment by robot-ch-test-poll2** [comment]
*This is an automated comment for commit bd5e4f42a635b30d49c8762c94e442ad910c36f7 with description of existing statuses. It's updated for the latest CI running*

[⏳ Click here](https://s3.amazonaws.com/clickhouse-test-reports/60337/bd5e4f42a635b30d49c8762c94e442ad910c36f7/ci_running.html) to open a full report in a separate page

<table>
<thead><tr><th>Check name</th><th>Description</th><th>Status</th></tr></thead>
<tbody>
<tr><td>CI running</td><td>A meta-check that indicates the running CI. Normally, it's in <b>success</b> or <b>pending</b> state. The failed status indicates some problems with the PR</td><td><a href="https://s3.amazonaws.com/clickhouse-test-reports/60337/bd5e4f42a635b30d49c8762c94e442ad910c36f7/ci_running.html">⏳ pending</a></td></tr>
<tbody>
</table>

File: docs/en/sql-reference/statements/insert-into.md [hunk 1]
```diff
@@ -230,7 +236,15 @@ Performance will not decrease if:
 - Data is added in real time.
 - You upload data that is usually sorted by time.
 
-It's also possible to asynchronously insert data in small but frequent inserts. The data from such insertions is combined into batches and then safely inserted into a table. To enable the asynchronous mode, switch on the [async_insert](../../operations/settings/settings.md#async-insert) setting. Note that asynchronous insertions are supported only over HTTP protocol, and deduplication is not supported for them.
+### Asynchronous inserts
+
+It is possible to asynchronously insert data in small but frequent inserts. The data from such insertions is combined into batches and then safely inserted into a table. To use asynchronous inserts, enable the [`async_insert`](../../operations/settings/settings.md#async-insert) setting. Note that asynchronous inserts are supported only over HTTP protocol and that deduplication is not supported for them.
```
Comment (alexey-milovidov) [inline docs/en/sql-reference/statements/insert-into.md:241]: They are supported in the native protocol as well, and deduplication is also supported.

File: docs/en/sql-reference/statements/insert-into.md [hunk 2]
```diff
@@ -230,7 +236,15 @@ Performance will not decrease if:
...
+
+Using `async_insert` or the [`Buffer` table engine](/en/engines/table-engines/special/buffer) results in results in additional buffering.
```
Comment (alexey-milovidov) [inline docs/en/sql-reference/statements/insert-into.md:243]:
```suggestion
Using `async_insert` or the [`Buffer` table engine](/en/engines/table-engines/special/buffer) results in additional buffering.
```

File: docs/en/sql-reference/statements/insert-into.md [hunk 3]
```diff
@@ -230,7 +236,15 @@ Performance will not decrease if:
...
+
+### Long-running inserts
+
+For long-running inserts, results will be visible after writing each [`max_insert_block_size`](/en/operations/settings/settings#max_insert_block_size) rows.
```
Comment (alexey-milovidov) [inline docs/en/sql-reference/statements/insert-into.md:247]: Add info about squashing.

File: docs/en/sql-reference/statements/insert-into.md (hunk 1 above)
Comment (justindeguzman) [reply docs/en/sql-reference/statements/insert-into.md:241]: Ok just to clarify, deduplication is supported for both native protocol and HTTP protocol? So I should just remove the following sentence:

>Note that asynchronous inserts are supported only over HTTP protocol and that deduplication is not supported for them.

File: docs/en/sql-reference/statements/insert-into.md (hunk 3 above)
Comment (justindeguzman) [reply docs/en/sql-reference/statements/insert-into.md]: Rewrote the section as the following:

> When you are inserting large amounts of data, ClickHouse will optimize write performance through a process called "squashing". Small blocks of inserted data in memory are merged and squashed into larger blocks before being written to disk. Squashing reduces the overhead associated with each write operation. In this process, inserted data will be available to query after ClickHouse completes writing each [`max_insert_block_size`](/en/operations/settings/settings#max_insert_block_size) rows.

File: docs/en/sql-reference/statements/insert-into.md (hunk 1 above)
Comment (qoega) [reply docs/en/sql-reference/statements/insert-into.md:241]: Enabled explicitly now:
https://clickhouse.com/docs/en/operations/settings/settings#async-insert-deduplicate

Dedup window in inserts and time:
https://clickhouse.com/docs/en/operations/settings/merge-tree-settings#replicated-deduplication-window-for-async-inserts
https://clickhouse.com/docs/en/operations/settings/merge-tree-settings#replicated-deduplication-window-seconds-for-async_inserts

File: docs/en/sql-reference/statements/insert-into.md (hunk 3 above)
Comment (alexey-milovidov) [reply docs/en/sql-reference/statements/insert-into.md]: This is not entirely correct, better not add this paragraph.

**Review by alexey-milovidov — APPROVED** [review]

**Pull request merged**